### ISSUE TITLE
setElementBoneMatrix API

### ISSUE THREAD

**Issue by Zangomangu** [multitheftauto/mtasa-blue#1887]
setElementBoneMatrix should accept a 4x4 lua table to keep it similar to getElementBoneMatrix and the other matrix functions (setElementMatrix/getElementMatrix). Right now it only accepts MTA matrix.

**Comment by Pirulax** [comment]
@saml1er what do you think?
 Should I make it consistent?

**Comment by ghost** [comment]
> 
> 
> @saml1er what do you think?
> Should I make it consistent?

Yes, that would be really nice. If you're going to write the PR, then please consider fixing setElementBonePosition. It takes a Vector3 instead of three floating points.

**Comment by Pirulax** [comment]
Thats fine, the parser will convert 3 float arguments to a single Vector, just like the arg reader does.

**Comment by ghost** [comment]
> 
> 
> Thats fine, the parser will convert 3 float arguments to a single Vector, just like the arg reader does.

But that's not what I want, I want it to take 3 floats.

**Comment by qaisjp** [comment]
> But that's not what I want, I want it to take 3 floats.

What do you mean? The old and new argument parser accepts three floats in place of a Vector defined on the C++ side.

**Comment by ghost** [comment]
> What do you mean? The old and new argument parser accepts three floats in place of a Vector defined on the C++ side.

You're right. It seems there was a bug in my script which was throwing Vector errors.

**Comment by Zangomangu** [comment]
> Should I make it consistent?

@Pirulax were you gonna look at this?
It would be nice to get it sorted out
